client/src/features/profile/setting/ProfileSetting.tsx

```diff
@@ -5,154 +5,190 @@ import SenpaiProfileSettingItems from './SenpaiProfileSettingItems';
 import KohaiProfileSettingItems from './KohaiProfileSettingItems';
 import axios from 'axios';
 import ToggleButton from '../../../common/components/ToggleButton';
+import { useAppSelector } from '../../../app/hook';
 
 const ProfileSetting: React.FunctionComponent<{ props?: any }> = () => {
-	const { pathname } = useLocation(); // used as a flag to see if it's senpai or kohai *returns as /profile/setting/senpai
-	const navigate = useNavigate();
-	const [isSenpai, setIsSenpai] = useState<undefined | boolean>(undefined);
-	const [inputs, setInputs] = useState<IProfileSetting>({
-		name: '',
-		publicEmail: '',
-		techStack: [],
-		description: '',
-	});
+  const myProfile = useAppSelector((state) => state.myProfile.myProfile);
+  const { pathname } = useLocation(); // used as a flag to see if it's senpai or kohai *returns as /profile/setting/senpai
+  const navigate = useNavigate();
+  const [isSenpai, setIsSenpai] = useState<undefined | boolean>(undefined);
+  const profileType = isSenpai ? 'senpaiProfile' : 'kohaiProfile';
+  const [inputs, setInputs] = useState<IProfileSetting>({
+    name: '',
+    publicEmail: '',
+    techStack: [],
+    description: '',
+    isActive: false,
+  });
 
-	useLayoutEffect(() => {
-		if (pathname === '/profile/setting/senpai') {
-			setIsSenpai(true);
-		} else {
-			setIsSenpai(false);
-		}
-	}, [pathname]);
+  useLayoutEffect(() => {
+    if (pathname === '/profile/setting/senpai') {
+      setIsSenpai(true);
+    } else {
+      setIsSenpai(false);
+    }
 
-	const handleOnChange = (
-		e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
-	) => {
-		const { name, value } = e.target;
-		if (name === 'techStack') {
-			const commaSpliceArr: string[] = value.split(',');
-			const returnedArr: string[] = [];
-			commaSpliceArr.map((ele) => returnedArr.push(ele.trim()));
-			setInputs({ ...inputs, techStack: returnedArr });
-		} else {
-			setInputs({ ...inputs, [name]: value });
-		}
-	};
+    if (Object.keys(myProfile).length !== 0) {
+      const publicEmail = myProfile.publicEmail
+        ? myProfile.publicEmail
+        : myProfile.email;
```

Comment:
We added the conditional here to see if there is `publicEmail` in `myProfile`. If not, we set an email that user used for login.